daemon/gluster-blockd.c

```diff
@@ -262,13 +267,108 @@ glusterBlockDParseArgs(int count, char **options)
   return 0;
 }
 
+
+static void
+gbMinKernelVersionCheck(void)
+{
+  struct utsname verStr = {'\0', };
+  char out[32] = {'\0', };
+  size_t vNum[VERNUM_BUFLEN] = {0, };
+  FILE *fp = NULL;
+  int i = 0;
+  char *tptr;
+
+
+  fp = popen(GB_DISTRO_CHECK, "r");
+  if (fp) {
+    size_t newLen = fread(out, sizeof(char), 32, fp);
+    if (ferror(fp)) {
+      LOG("mgmt", GB_LOG_ERROR, "fread(%s) failed: %s",
+          GB_DISTRO_CHECK, strerror(errno));
+      goto fail;
+    }
+    out[newLen++] = '\0';
+    tptr = strchr(out,'\n');
+    if (tptr) {
+      *tptr = '\0';
+    }
+  } else {
+    LOG("mgmt", GB_LOG_ERROR, "popen(%s): failed: %s",
+        GB_DISTRO_CHECK, strerror(errno));
+    goto fail;
+  }
+
+  if (uname(&verStr) != 0) {
+    LOG("mgmt", GB_LOG_ERROR, "uname() failed: %s", strerror(errno));
+    goto fail;
+  }
+
+  tptr = verStr.release;
+  while (*tptr) {
+    if (isdigit(*tptr)) {
+      vNum[i] = strtol(tptr, &tptr, 10);
+      i++;
+    } else if (isalpha(*tptr)) {
+      break;
+    } else {
+      tptr++;
+    }
+
+    if (i >= VERNUM_BUFLEN) {
+      break;
+    }
+  }
+
+  if (strstr(out, "fedora")) {
+    tptr = "4.12.0-1"; /* Minimum recommended fedora kernel version */
+    if (KERNEL_VERSION(vNum[0], vNum[1], vNum[2]) < KERNEL_VERSION(4, 12, 0)) {
+      goto out;
+    } else if (KERNEL_VERSION(vNum[0], vNum[1], vNum[2]) == KERNEL_VERSION(4, 12, 0)) {
+      if (KERNEL_VERSION(vNum[3], vNum[4], vNum[5]) < KERNEL_VERSION(1, 0, 0)) {
+        goto out;
+      }
+    }
+  } else if (strstr(out, "rhel")) {
+    tptr = "3.10.0-862.11.1"; /* Minimum recommended rhel kernel version */
+    if (KERNEL_VERSION(vNum[0], vNum[1], vNum[2]) < KERNEL_VERSION(3, 10, 0)) {
+      goto out;
+    } else if (KERNEL_VERSION(vNum[0], vNum[1], vNum[2]) == KERNEL_VERSION(3, 10, 0)) {
+      if (KERNEL_VERSION(vNum[3], vNum[4], vNum[5]) < KERNEL_VERSION(862, 11, 1)) {
+        goto out;
+      }
+    }
+  } else {
+    LOG("mgmt", GB_LOG_INFO, "Distro %s. Skipping kernel version check.", out);
+  }
+
+  LOG("mgmt", GB_LOG_INFO, "Distro %s. Current kernel version: '%s'.",
+      out, verStr.release);
+
+  return;
```

Comment:
Brfore 'return' here, shouldn't we close the fp ?

Reply:
You got it. Will refresh.